### ISSUE TITLE
Documentation improvements

### ISSUE THREAD

**Issue by grcninja** [secynic/ipwhois#123]
So I'm struggling here to access the value in "address.

Here's the Example I'm following from the documentation
'objects': {
    u'ABUSE5250-ARIN': {
        'contact': {
            'address': [{
                'type': None,
                'value': u'1600 Amphitheatre Parkway\nMountain View\nCA\n
                    94043\nUNITED STATES'

Here's what I have written:  
        addr = '185.94.189.144'
        obj = IPWhois(addr, timeout=20)
        results = obj.lookup_rdap(depth=1)
        asn = tab_add(newline_clean(results['asn']))
        asn_country_code = tab_add(newline_clean(results['asn_country_code']))
        network_cidr = tab_add(newline_clean(results['network']['cidr']))
        pp(results['objects'])
        contact_address = tab_add(newline_clean(results['objects']['contact']['address']['value']))

now I wrote my contact_address line based on the HTML tables showing the relationship between Objects Dictionary and Contact Dictionary.  Yet when I go back and look at the Example I see a level between them (u'ABUSE5250-ARIN': ) but I do not see a reference to this is the Object Dictionary table.

Can you please tell me what I am missing here?  It is probably a mistake that I am making between interpreting what you meant and what I'm reading.  Simply fixing this line for me or giving an actual example of how to extract this value that is buried further down than Hades would be sooooooooo much appreciated.

**Comment by secynic** [comment]
The key part you are missing is that objects is a dictionary of nested object dictionaries. In the description for objects: "Keys are the object names with values as Objects Dictionary." Basically,
there can be multiple objects for a result, especially when the depth value is larger.

I do need to fix the description for Objects Contact - address, email, and phone. Those fields are actually lists of dictionaries (since you can have multiple address, email, and phone entries per object).

So, to get the address for ABUSE5250-ARIN, your code would be something like:
`contact_address = tab_add(newline_clean(results['objects']['ABUSE5250-ARIN']['contact']['address'][0]['value']))`

That is simplified for the sake of the example. address, email, phone could be None:
```
tmp = results['objects']['ABUSE5250-ARIN']['contact']['address']
if tmp is not None:
    tab_add(newline_clean(tmp[0]['value']))
```

The above still just gets you the first contact address result. Depending on how you are displaying your data, you may want to display all the listed addresses. To do that, you will need to change how you are displaying your data, and iterate the address, contact, phone:
```
tmp = results['objects']['ABUSE5250-ARIN']['contact']['address']
contact_addresses = []
if tmp is not None:
    for addr_dict in tmp:
        contact_addresses.append(addr_dict['value'])
```

Does that make sense? I would greatly appreciate any help or suggestions on how to simplify the documentation for this.

**Comment by grcninja** [comment]
That is actually very helpful.  What would be even more helpful is if there's some way to know what results could be expected back.  For example, am I guaranteed to get 'ABUSE5250' back every time?  If not, then I'll need to loop through the objects as well. Correct?  I think the hardest thing is that there aren't a lot of examples like you just gave me.  After some sleep and some coffee I came to realize this morning that I'd need something like what you gave me above.

Please know that I think you have made a VERY VERY awesome package here, and please take nothing I say as a criticism.  You've done great.  More code example snippets would be the one thing I think could be helpful.

**Comment by secynic** [comment]
Haha, thank you.

Yes, since objects will be different for each IP, you should always loop that dictionary:
```
for object_key, object_dict in results['objects'].items():
    # Do stuff
```

0.13.0 was my first attempt at documenting this beast. I assure you that more examples and easier to understand docs are being worked on. I will leave this open to track those improvements.

**Comment by grcninja** [comment]
Here if this helps you can steal it and use it...I haven't tested it to make sure I don't have a syntax error, so check it first.  Should get me the first value returned for each item.  I only need one for my purposes

        for k in results['objects']:
            tmp_add = results['objects'][k]['contact']['address'][0]['value']
            if tmp_add is not None:
                contact_address = tab_add(newline_clean(tmp_add[0]['value']))
            tmp_em = results['objects'][k]['contact']['email'][0]['value']
            if tmp_em is not None:
                contact_email = tab_add(newline_clean(tmp_em[0]['value']))
            tmp_ph = results['objects'][k]['contact']['phone'][0]['value']
            if tmp_ph is not None:
                contact_phone = tab_add(newline_clean(tmp_ph[0]['value']))
            tmp_nm = results['objects'][k]['contact']['phone'][0]['value']
            if tmp_nm is not None:
                contact_name = tab_add(newline_clean(tmp_nm[0]['value']))
            tmp_ti = results['objects'][k]['contact']['title'][0]['value']
            if tmp_ti is not None:
                contact_title = tab_add(newline_clean(tmp_ti[0]['value']))
            tmp_ro = results['objects'][k]['contact']['role'][0]['value']
            if tmp_ro is not None:
                contact_role = tab_add(newline_clean(tmp_ro[0]['value']))

**Comment by secynic** [comment]
Some changes to above:
```
for k in results['objects']:
        tmp_add = results['objects'][k]['contact']['address']
        if tmp_add is not None:
            contact_address = tab_add(newline_clean(tmp_add[0]['value']))
        tmp_em = results['objects'][k]['contact']['email']
        if tmp_em is not None:
            contact_email = tab_add(newline_clean(tmp_em[0]['value']))
        tmp_ph = results['objects'][k]['contact']['phone']
        if tmp_ph is not None:
            contact_phone = tab_add(newline_clean(tmp_ph[0]['value']))
        tmp_nm = results['objects'][k]['contact']['name']
        if tmp_nm is not None:
            contact_name = tab_add(newline_clean(tmp_nm))
        tmp_ti = results['objects'][k]['contact']['title']
        if tmp_ti is not None:
            contact_title = tab_add(newline_clean(tmp_ti))
        tmp_ro = results['objects'][k]['contact']['role']
        if tmp_ro is not None:
            contact_role = tab_add(newline_clean(tmp_ro))
```

Note I removed [0]['value'] from tmp variables before None check. I also removed it from name/title/role, as those are string outputs, not lists/nested dicts.

**Comment by grcninja** [comment]
Last one - so that I can grab that identifier (the u'$something' like ABUSE5250-ARIN) is this right?

            tmp_id = results['objects'][k]
            if tmp_id is not None:
                entry_id = results['objects'][k]

**Comment by secynic** [comment]
In your example, k would be that value.

To simplify, try:
```
if results['objects'] is not None:
    for object_key, object_dict in results['objects'].items():
        entry_id = object_key
        # Do stuff
```

Above, object_key is what you want.

P.S. I am working and unable to test any of these examples I am giving you. Hopefully no errors.

**Comment by secynic** [comment]
\+ Fix document section linking. Example of issue: https://secynic.github.io/ipwhois/RDAP.html

**Comment by secynic** [comment]
+ Add version info in examples for elasticsearch, kibana, redis
+ Need to validate if the examples are working with the latest versions, and fix if necessary (create new issue).

**Comment by secynic** [comment]
Merged in #139. See note above about creating #138 for next release.
